### ISSUE TITLE
Ensemble loss calculation

### ISSUE THREAD

**Issue by harveyp123** [samuelstanton/gnosis#17]
On the line
https://github.com/samuelstanton/gnosis/blob/1b81adcbcd1566d7ea6dda1594f276d64950c998/gnosis/distillation/classification.py#L39
n_teachers = len(teacher_logits)
I think the first dimension is batch size? The teacher_logits should have input with shape (batch size, num_teachers, num_class)

**Comment by harveyp123** [comment]
n_teachers = teacher_logits.shape[1] shall do the job

**Comment by samuelstanton** [comment]
it's just a scaling term so shouldn't affect results much, but nice catch.

**Comment by samuelstanton** [comment]
since it shouldn't affect reproducibility I've pushed the fix
https://github.com/samuelstanton/gnosis/commit/7a40c5d15ecdf11d72e44fc3e6946b79c18e0c4e
